File: src/main/java/com/itzk/SmartEmploymentPlatform/pojo/entry/Interview.java
package com.itzk.SmartEmploymentPlatform.pojo.entry;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 面试邀约实体类
 * 对应数据库中的 interview_invitations 表
 */
@Data
public class Interview {

    /** 面试邀约ID */
    private Long id;

    /** 投递记录ID（关联applications表） */
    private Long applicationId;

    /** 公司ID（冗余，便于查询） */
    private Long companyId;

    /** 职位ID（冗余） */
    private Long jobId;

    /** 求职者用户ID（冗余） */
    private Long userId;

    /** 发送邀约的HR用户ID */
    private Long hrUserId;

    /** 面试时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime interviewTime;

    /** 面试地点（可为线上链接或详细地址） */
    private String interviewLocation;

    /** 地点类型：0=线下 1=线上 */
    private Integer locationType;

    /** 联系人 */
    private String contactPerson;

    /** 联系电话 */
    private String contactPhone;

    /** HR备注（如需要携带的材料等） */
    private String remark;

    /** 状态：0=待确认 1=接受 2=拒绝 3=已过期 4=已取消 */
    private Integer status;

    /** 求职者回复备注 */
    private String candidateRemark;

    /** 求职者回复时间 */
    private LocalDateTime responseTime;

    /** 邀约创建时间 */
    private LocalDateTime createdAt;

    /** 最后更新时间 */
    private LocalDateTime updatedAt;
}

File: frontend/src/api/hr/interview.js
import request from "@/utils/request";

/**
 * 获取面试列表（HR端）
 */
export function getInterviewList() {
  return request({ url: "/hr/interviews/list", method: "get" });
}

/**
 * 更新面试状态（复用 /hr/interview/status）
 * 后端根据 status 判断：待定→仅更新、面试成功→同步更新申请为已录用、拒绝→同步更新申请为不合适
 * @param {number} interviewId - 面试ID
 * @param {number} status - 目标状态码：2=待定 3=面试成功 4=拒绝 5=过期 6=取消
 * @param {string|null} remark - 状态变更备注
 */
export function updateInterviewStatus(interviewId, status, hrRemark) {
  return request({ url: "/hr/interviews/status", method: "put", data: { id: interviewId, status, hrRemark } });
}

/**
 * 获取面试详情
 * @param {number} interviewId - 面试ID
 */
export function getInterviewDetail(interviewId) {
  return request({ url: `/hr/interviews/${interviewId}`, method: "get" });
}

/**
 * 创建/更新面试邀约
 * 后端根据 data.id 是否存在决定：有id→更新，无id→插入
 * @param {Object} data - { id?, applicationId, userId, jobId, status, interviewTime, locationType, interviewLocation, contactPerson, contactPhone, hrRemark }
 */
export function createInterviewInvitation(data) {
  return request({ url: "/hr/resumes/interview", method: "post", data });
}

/**
 * 根据jobId+userId查询已有面试邀约（点击邀请面试时回显用）
 * @param {number} jobId
 * @param {number} userId
 */
export function getInterviewByJobAndUser(jobId, userId) {
  return request({ url: "/hr/interviews/query", method: "get", params: { jobId, userId } });
}

